controllers/k8ssandra/cleanup.go

```diff
@@ -478,3 +478,47 @@ func (r *K8ssandraClusterReconciler) deleteCronJobs(
 	}
 	return
 }
+
+func (r *K8ssandraClusterReconciler) deleteCassandraDatacenter(
+	ctx context.Context,
+	dcTemplate k8ssandraapi.CassandraDatacenterTemplate,
+	namespace string,
+	remoteClient client.Client,
+	logger logr.Logger,
+) (hasErrors bool, earlyReturn *result.ReconcileResult) {
+	dcKey := client.ObjectKey{Namespace: namespace, Name: dcTemplate.Meta.Name}
+	dc := &cassdcapi.CassandraDatacenter{}
+	err := remoteClient.Get(ctx, dcKey, dc)
+	if err != nil {
+		// If the DC is not found, it means it has already been deleted.
+		if !errors.IsNotFound(err) {
+			logger.Error(err, "Failed to get CassandraDatacenter for deletion",
+				"CassandraDatacenter", dcKey, "Context", dcTemplate.K8sContext)
+			hasErrors = true
+		}
+		return
+	}
+
+	// Remove finalizer before deletion
+	if controllerutil.ContainsFinalizer(dc, k8ssandra.K8ssandraClusterFinalizer) {
+		patch := client.MergeFrom(dc.DeepCopy())
+		controllerutil.RemoveFinalizer(dc, k8ssandra.K8ssandraClusterFinalizer)
+		if err := remoteClient.Patch(ctx, dc, patch); err != nil {
+			logger.Error(err, "Failed to remove finalizer from CassandraDatacenter", "CassandraDatacenter", dcKey, "Context", dcTemplate.K8sContext)
+			hasErrors = true
```

Comment:
issue: The error is lost here (and other places in this method that just return hasErrors).